spec/lang/step/intrinsics.md

```diff
@@ -307,10 +307,11 @@ impl<M: Memory> Machine<M> {
             throw_ub!("invalid number of arguments for `Spawn` intrinsic");
         }
 
-        let Value::Ptr(Pointer { thin_pointer: ptr, metadata: None }) = arguments[0].0 else {
+        let fn_ptr = arguments[0].0;
+        if !(matches!(fn_ptr, Value::Ptr(Pointer { metadata: None, .. }))) {
             throw_ub!("invalid first argument to `Spawn` intrinsic: not a thin pointer");
         };
```

Comment:
This should not be needed any more, `fn_from_ptr` should be checking everything there is to check.